### PR TITLE
Clear out authorization header from error logging

### LINKED ISSUE #11: Hide token in error messaging
As an integrade QE I would like to not have the authentication token shown in the 4xx or 5xx response exception messages. The output should be sanitized since we will sometimes be running against public instances of cloudigrade!

---

## Acceptance Criteria

- [ ] Verify that when an error message is raised by the API client, it does not include the token used.

Example:
```
HTTPError: 
============================================================

The request you made received a status code that indicates
an error was encountered. Details about the request and the
response are below.

============================================================
request path : '/api/v1/account/'

request body : (b'{"resourcetype": "AwsAccount", "account_arn": "blah"}')

request headers : {'User-Agent': 'python-requests/2.18.4', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive', 'Authorization': 'Token THISNEEDSTOBESANITIZED', 'Content-Length': '108', 'Content-Type': 'application/json'}

response code : 500

text_error_message : '<h1>Server Error (500)</h1>'
============================================================
```
I think it would be good to show that there was an 'Authorization' part of the header, just need to strip out the actual token.

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #11

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -97,6 +98,9 @@ ENV/
 # Rope project settings
 .ropeproject
 
+# Visual Code settings
```
Comment (elyezer) [inline .gitignore]: I would recommend adding personal editor to the global gitignore so we don't end up adding one ignore by editor depending on the personal taste of each one.

File: .gitignore [hunk 2]
```diff
@@ -89,6 +89,7 @@ celerybeat-schedule
 .venv
 venv/
 ENV/
+Pipfile
```
Comment (elyezer) [inline .gitignore]: Observation: we may start using Pipfile in the future since this is the new standard.

File: integrade/api.py [hunk 3]
```diff
@@ -37,14 +39,19 @@ def raise_error_for_status(response):
             response_message = 'text_error_message : {}'.format(
                 pformat(response.text))
 
+        error_headers = {}
+        if isinstance(response.request.headers, dict):
```
Comment (elyezer) [inline integrade/api.py]: This won't be different from a dict, only on mocking. I would rather update the test to properly mock a response object by having a dict assigned to response.request.headers.

Comment (elyezer) [reply integrade/api.py]: Also maybe worth adding a test to check if the response authorization header is being set to 8 * '*'

**Review by elyezer — CHANGES_REQUESTED** [review]

File: integrade/api.py (hunk 3 above)
Comment (elyezer) [reply integrade/api.py]: Here are the changes I would have on this PR

```diff
diff --git a/integrade/api.py b/integrade/api.py
index eadd5c3..2697b17 100644
--- a/integrade/api.py
+++ b/integrade/api.py
@@ -39,11 +39,9 @@ def raise_error_for_status(response):
             response_message = 'text_error_message : {}'.format(
                 pformat(response.text))
 
-        error_headers = {}
-        if isinstance(response.request.headers, dict):
-            error_headers = response.request.headers.copy()
-            if error_headers.get(AUTHORIZATION_HEADER) is not None:
-                error_headers[AUTHORIZATION_HEADER] = '*' * 8
+        error_headers = response.request.headers.copy()
+        if AUTHORIZATION_HEADER in error_headers:
+            error_headers[AUTHORIZATION_HEADER] = '*' * 8
         error_msgs += '\n\n'.join(
             [
                 'request path : {}'.format(pformat(
diff --git a/tests/test_api.py b/tests/test_api.py
index 08ec9c1..fb8ef10 100644
--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -30,7 +30,7 @@ def mock_request():
     mock_request = mock.Mock(
         body='{"Test Body"}',
         path_url='/example/path/',
-        headers='{"Test Header"}',
+        headers={'Authorization': 'authorizationkey'},
         text='Some text',
     )
     return mock_request
@@ -60,6 +60,7 @@ def bad_response_valid_json():
 def bad_response_invalid_json():
     """Return a mock response with a 404 status code but with bad json."""
     mock_response = mock.Mock(status_code=404, text='<No JSON!>')
+    mock_response.request = mock_request()
     mock_response.json = Mock()
     mock_response.json.side_effect = JSONDecodeError(
         msg='no json',
```

File: .gitignore (hunk 1 above)
Comment (kholdaway) [reply .gitignore]: 👍

**Comment by kholdaway** [comment]
@elyezer take another look

**Review by elyezer — APPROVED** [review]

**Pull request merged**